includes/Services/InstaMigrateService.php

```diff
@@ -26,82 +19,55 @@ class InstaMigrateService {
 	private $insta_api_key = '';
 
 	/**
-	 * Retry count
+	 * Tracker class instance.
 	 *
-	 * @var int $count
+	 * @var Tracker $tracker
 	 */
-	private $count = 0;
+	private $tracker;
 
 	/**
 	 * Set required api keys for insta to initiate the migration
 	 */
 	public function __construct() {
-		$encrypt             = new Encryption();
-		$this->insta_api_key = $encrypt->decrypt( get_option( 'newfold_insta_api_key', false ) );
-		if ( ! $this->insta_api_key ) {
-			$this->insta_api_key = UtilityService::get_insta_api_key( BRAND_PLUGIN );
-			update_option( 'newfold_insta_api_key', $encrypt->encrypt( $this->insta_api_key ) );
-		}
+		$this->tracker = new Tracker();
+		$this->tracker->reset_track_file();
+		$instawp_get_key_step = new GetInstaWpApiKey( $this->tracker );
+		$instawp_get_key_step->set_status( 'running' );
+		$this->insta_api_key = $instawp_get_key_step->get_api_key();
 	}
 
 	/**
 	 * Install InstaWP plugin
 	 */
 	public function install_instawp_connect() {
-		if ( ! function_exists( 'get_plugins' ) || ! function_exists( 'get_mu_plugins' ) ) {
-			require_once ABSPATH . 'wp-admin/includes/plugin.php';
-		}
-		// Install and activate the plugin
-		if ( ! is_plugin_active( sprintf( '%1$s/%1$s.php', $this->connect_plugin_slug ) ) ) {
-			$params    = array(
-				array(
-					'slug'     => 'instawp-connect',
-					'type'     => 'plugin',
-					'activate' => true,
-				),
-			);
-			$installer = new Installer( $params );
-			$response  = $installer->start();
-		}
-
-		// Connect the website with InstaWP server
-		if ( empty( Helper::get_api_key() ) || empty( Helper::get_connect_id() ) ) {
-			$api_key          = Helper::get_api_key( false, $this->insta_api_key );
-			$connect_response = Helper::instawp_generate_api_key( $api_key, '', false );
+		$install_activate = new InstallActivateInstaWp( $this->tracker );
+		$install_activate->set_status( 'running' );
+		$installed_activated = $install_activate->install();
```

Comment:
Similar to my previous comment, all we need to do is call the failed function to proceed or throw an error.

This should be implemented consistently across all steps, ensuring:

1. Auto run execution upon instantiation
2. Auto retries if needed
3. Auto status updates

The InstaMigrateService shouldn't have to deal with step-specific functions like:
```
$connectToInstaWp->connect();
$instawp_get_key_step->get_api_key();
```
Instead, it should only instantiate the step classes based on what passed or failed.

Once instantiated, the step should automatically:

1. Run its logic
2. Retry if necessary
3. Set its status (running, failed, completed)

Then, the service can simply check the status and proceed accordingly, without requiring custom logic for each step.